Dockerfile

```diff
@@ -1,24 +1,29 @@
-FROM golang:1.12-alpine
+FROM golang:1
 
 ENV PROJECT=coreos-version-checker
-COPY . /${PROJECT}-sources/
+ENV ORG_PATH="github.com/Financial-Times"
+ENV REPO_PATH="${ORG_PATH}/${PROJECT}"
+ENV SRC_FOLDER="${GOPATH}/src/${REPO_PATH}"
 
-RUN apk --no-cache --virtual .build-dependencies add git curl \
-  && ORG_PATH="github.com/Financial-Times" \
-  && REPO_PATH="${ORG_PATH}/${PROJECT}" \
+COPY . ${SRC_FOLDER}
+WORKDIR ${SRC_FOLDER}
+
+RUN curl https://raw.githubusercontent.com/golang/dep/master/install.sh | sh \
+  && $GOPATH/bin/dep ensure -vendor-only \
   && mkdir -p $GOPATH/src/${ORG_PATH} \
   # Linking the project sources in the GOPATH folder
   && ln -s /${PROJECT}-sources $GOPATH/src/${REPO_PATH} \
-  && cd $GOPATH/src/${REPO_PATH} \
+  && cd ${SRC_FOLDER} \
   && echo "Fetching dependencies..." \
-  && curl https://raw.githubusercontent.com/golang/dep/master/install.sh | sh \
-  && $GOPATH/bin/dep ensure -vendor-only \
   && go build  \
-  && mv ${PROJECT} /${PROJECT} \
-  && apk del .build-dependencies \
-  && rm -rf $GOPATH /var/cache/apk/*
+  && mv ${PROJECT} /${PROJECT}
+
+# Multi-stage build - copy certs and the binary into the image
+FROM scratch
 
 WORKDIR /
+COPY --from=0 /etc/ssl/certs/ca-certificates.crt /etc/ssl/certs/
+COPY --from=0 /${SRC_FOLDER} /
```

Comment:
Shouldn't we just copy only the application? And not the source.

Reply:
You are right, I fixed it